glms.qmd

```diff
@@ -0,0 +1,115 @@
+---
+title: ""
+number-sections: true
+format: 
+  html:
+    embed-resources: true
+    code-tools: true
+  pdf: default
+editor_options: 
+  chunk_output_type: console
+execute:
+  echo: true
+  eval: true
+  warning: false
+  message: false
+---
+
+```{r}
+library(tidyverse)
+library(moderndive)
+library(gapminder)
+library(sjPlot)
+library(stats)
+library(jtools)
+library(dplyr)
+library(knitr)
+library(janitor)
+library(MASS)
+library(sjPlot)
+```
+
+Data wrangling and subseting
+
+```{r}
+#data wrangling and subseting
+data = read.csv('dataset14.csv')
+mapping = c('Poor' = 0, 'Good' = 1)
+data$Qualityclass = mapping[data$Qualityclass]
+data$country_of_origin = as.factor(data$country_of_origin)
+data$harvested = as.factor(data$harvested)
+data = data %>%
+  filter(country_of_origin == 'Brazil' | country_of_origin == 'Colombia')
+```
+
+```{r}
+table(data$country_of_origin)
+```
+
+## Not removing variables
+
+Here we use all variables to fit a full logistic regression model:
+
+```{r}
+full_model = glm(formula = data$Qualityclass ~ ., family = binomial(link = 'logit'),data = data)
+```
+
+```{r}
+full_model %>%
+  summary()
+```
+
+AIC:164.97
+
+```{r}
+table(data$harvested)
+```
+
```

Comment:
stepAIC(full_model) maybe a good way to have initial impression on varibles selection. Could you add this code and some comments there?
(Otherwise, we can not ensure that delete other significant varible can not cause a lower AIC.)
And based on the summary provided by stepAIC, we can directly try to delete variables 'category_two_defects' or 'altitude_mean_meters' without any other concern.

Reply:
Reasonable suggestions! I have adopted and changed my code.